src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFileProvider.cs

```diff
@@ -161,7 +161,13 @@ internal PhysicalFilesWatcher CreateFileWatcher()
             string root = PathUtils.EnsureTrailingSlash(Path.GetFullPath(Root));
 
             // When both UsePollingFileWatcher & UseActivePolling are set, we won't use a FileSystemWatcher.
-            FileSystemWatcher watcher = UsePollingFileWatcher && UseActivePolling ? null : new FileSystemWatcher(root);
+            FileSystemWatcher watcher = UsePollingFileWatcher && UseActivePolling ? null :
+#if NETCOREAPP
+                OperatingSystem.IsBrowser() ? throw new PlatformNotSupportedException(SR.Format(SR.FileSystemWatcher_PlatformNotSupported, typeof(FileSystemWatcher))) : new FileSystemWatcher(root);
+#else
```

Comment:
Won't this still throw PNSE when creating a `PhysicalFileProvider` on browser? Shouldn't the logic be something like:


```C#
(UsePollingFileWatcher && UseActivePolling) || OperatingSystem.IsBrowser() ? null : ...
```

Reply:
Ah I see, so we should always fallback to use polling on browser. That makes sense to me.

Reply:
FYI: PR https://github.com/dotnet/runtime/pull/56257 addresses this in https://github.com/dotnet/runtime/pull/56257/commits/276b0089ffbbf571db09bf8f8025b5a6cf81751c.